WIP: Solitaire

IgorMacGregor opened the pull request:
Hi guys, 
Here is my up to date version of the Solitaire view (with Hooks !), I will now focus on the logic of the game, but I will need some time to understand how to deal with the different events of a game. Sorry, I had to create a new pull request, I was struggling hard with Git for the last two days :'(
Have a good day !

ShizukuIchi commented:
Hi @IgorMacGregor! You should change some of  your lint rules such as single quote, comma-dangle.

IgorMacGregor commented:
Hi, I'm still working on the solitaire and I would maybe need some advice up to this point. Here are the new changes:
- Added drag functionality on cards, which is a bit slow. How could I make it faster so that the card follows exactly the cursor ?
- Added drop functionality on bottom stacks. Here, I have been stuck for three days as I can only move one card, and when I move another one the global state forgets the previous move ... (I have disabled the randomization of the game so that it is easier to test two moves)
- Also, to make the drop event fired, I needed to lower the z-index of the moving card, whereas in the real solitaire game it is ahead of the rest of the game, and I'm still thinking to how to deal with that.

If you have any advice/solution to my problems tell me ! 
Thanks !!

IgorMacGregor commented:
Hey, I have made some changes which allow in a strange way to make several moves, I think it was because of a closure issue but now it works :) I have also worked on making deck and topslots functional. 
From now on, I think my React skills are too limited, particularly for making several cards move, so if someone wants to work on the Solitaire some help would be much needed :)
There is still some lags problems and the z-index issue. Thanks !
